src/test/java/tatracker/model/module/UniqueModuleListTest.java

```diff
@@ -0,0 +1,190 @@
+package tatracker.model.module;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static tatracker.testutil.Assert.assertThrows;
+import static tatracker.testutil.module.TypicalModules.CS2103T;
+import static tatracker.testutil.module.TypicalModules.CS2103T_COPY;
+import static tatracker.testutil.module.TypicalModules.CS3243;
+import static tatracker.testutil.module.TypicalModules.NO_GROUPS;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+public class UniqueModuleListTest {
+
+    private final UniqueModuleList uniqueModuleList = new UniqueModuleList();
+
+    @Test
+    public void contains_nullModule_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueModuleList.contains(null));
+    }
+
+    @Test
+    public void contains_moduleNotInList_returnsFalse() {
+        assertFalse(uniqueModuleList.contains(CS2103T));
+    }
+
+    @Test
+    public void contains_moduleInList_returnsTrue() {
+        uniqueModuleList.add(CS2103T);
+        assertTrue(uniqueModuleList.contains(CS2103T));
+    }
+
+    @Test
+    public void contains_moduleWithSameIdentityFieldsInList_returnsTrue() {
+        uniqueModuleList.add(CS2103T);
+        assertTrue(uniqueModuleList.contains(CS2103T_COPY));
+    }
```

Comment:
Invalid test cases. Can choose between creating the module list each time, or use a @BeforeEach method to reset the unique list. Else in the list will be carried over to the other test cases

Reply:
idk why, but this works. I legit copied the UniqueStudentList test cases, edited them and added more to do this.

Reply:
Adding before each introduced some strange errors that I didn't understand so I have decided not to use it.